test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -92,6 +92,37 @@ var _ = BeforeSuite(func() {
 })
 
 var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", func() {
+	Describe("when docker service is stopped", func() {
+		It("node should become ready again in a reasonable amount of time", func() {
+			kubeConfig, err := GetConfig()
+			Expect(err).NotTo(HaveOccurred())
+			master := fmt.Sprintf("%s@%s", eng.ExpandedDefinition.Properties.LinuxProfile.AdminUsername, kubeConfig.GetServerName())
+			nodeList, err := node.GetReady()
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, node := range nodeList.Nodes {
```

Comment:
should this be for all nodes? ie both Linux and Windows

Reply:
I originally intended to only do it for Windows nodes but I changed my mind on that, I think the Kubelet should be able to recover for both Windows and Linux node even if the docker is service is stopped. If the Linux nodes are giving trouble I'll switch it back to Windows only for now.